pkg/policy/l4.go

```diff
@@ -143,22 +143,21 @@ func (l7 L7DataMap) addRulesForEndpoints(rules api.L7Rules, endpoints []api.Endp
 	}
 }
 
-// CreateL4IngressFilter creates an L4Filter for the specified api.PortProtocol in
-// the ingress direction for a particular protocol.
-// This L4Filter will only apply to endpoints covered by `fromEndpoints`.
-// `rule` allows a series of L7 rules to be associated with this L4Filter.
-func CreateL4IngressFilter(fromEndpoints api.EndpointSelectorSlice, rule api.PortRule, port api.PortProtocol,
-	protocol api.L4Proto, ruleLabels labels.LabelArray) L4Filter {
+// CreateL4Filter creates a filter for L4 policy that applies to the specified
+// endpoints and port/protocol, with reference to the original rules that the
+// filter is derived from. This filter may be associated with a series of L7
+// rules via the `rule` parameter.
+func CreateL4Filter(peerEndpoints api.EndpointSelectorSlice, rule api.PortRule, port api.PortProtocol,
+	protocol api.L4Proto, ruleLabels labels.LabelArray, ingress bool) L4Filter {
```

Comment:
Any chance you can make `ingress bool` a `TrafficDirection` type with two possible values? Calls to this function will then not have a mysterious boolean parament but we could also drop the ingress/egress wrappers.

Reply:
hrm, we have a `TrafficDirection` type, but it's specific to the BPF datapath at the moment. Are you suggesting introducing another such type for this layer? /cc @ianvernon

Reply:
:/ I would prefer to reuse types if possible, but it's unlikely to be safe in this case. Perhaps it's out of scope since this is basically a bike-shed type thing on where such a thing should live and what it should be called.
Ultimately, my hope was to have a little less code. As it stands, it's readable since the Ingress/Egress is in the wrappers.

Reply:
very keen! I might prefer the direction earlier in the parameter list but I'm not sure if a caller considers it the most important/global property of the filter (I do but I also asked you to expose the parameter directly so I am biased).

Reply:
Putting `TrafficDirection` in the BPF datapath layer was a mistake on my part. IMO we can have it as its own little package? It'd make it much more portable and reusable.

Reply:
This seems like a reasonable suggestion, but it's getting further and further from what I'm trying to achieve so I intend to just drop this for now.

Reply:
Sure, that's fine.